### ISSUE TITLE
Allow restriction of preventOverflow to a single axis

### ISSUE THREAD

**Issue by tyscorp** [popperjs/popper.js#344]
I'd like the ability to force preventOverflow to only act upon a single axis. I'm currently overriding the behaviour like so:

```js
{
  modifiers: {
    saveYPosition: {
      enabled: true,
      fn(data, options) {
        data.storedTop = data.offsets.popper.top;

        return data;
      },
      order: 250
    },
    restoreYPosition: {
      enabled: true,
      fn(data, options) {
        data.offsets.popper.top = data.storedTop;

        return data;
      },
      order: 350
    }
  }
}
```

...but it would be nice to have this built-in to the modifier as an option, with both axis being enabled by default. I'm not sure whether it would be more useful to be able to restrict based on the `primary`/`secondary` axes or the `x`/`y` axes explicitly. Maybe both?

**Comment by FezVrasta** [comment]
There is other work pending on the preventOverflow modifier (#230).

I think it's a good candidate for a major overhaul to release in v2.

If you'd like to work on this I think it would be pretty effective if we could have a new folder under `packages` named something like `preventOverflowEnhanced` so we can work on it and break APIs as much we need and people can opt-in if they want to try it while we work on it.

Once it's ready we can replace the built in one with the new one.

How does it sounds?

**Comment by ernestp** [comment]
@FezVrasta Have any improvement since? 
I tried preventOverflow priority: ['bottom', 'top'], but still got placement at right
